plugins/net_plugin/net_plugin.cpp

```diff
@@ -3508,11 +3504,13 @@ namespace eosio {
 
       go_away_reason reason = fatal_other;
       try {
+         my_impl->dispatcher->add_peer_block( blk_id, c->connection_id );
```

Comment:
what about moving this line after `try{}catch()` ? In that case we don't need to have `rm_peer_block` at all.

Reply:
It has to be before `accept_block` because `accept_block` signals the sending of the block out to all the peers.

Reply:
If the block did not validate or it asserted there is no reason to ever try the block again. If it was unlinkable that could be because it has not received the prior block yet for some reason